libgadget/slotsmanager.c

```diff
@@ -532,7 +527,7 @@ slots_check_id_consistency()
     for(i = 0; i < PartManager->NumPart; i++) {
         if(!SLOTS_ENABLED(P[i].Type)) continue;
 
-        if(P[i].PI >= SlotsManager->info[P[i].Type].size) {
+        if(P[i].PI >= (unsigned int) SlotsManager->info[P[i].Type].size) {
```

Comment:
Why didn't we make .size unsigned?

Reply:
Maybe PI shall be signed? I think we do set it to (-1) in some occasions.

Reply:
NumPart is signed too, so making PI signed would work. I'll try that.